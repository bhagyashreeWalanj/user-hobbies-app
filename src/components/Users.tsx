import * as React from 'react'
import { useState, useEffect } from 'react'
import styles from './../styles/Users.module.scss'
import { IUser } from './../interfaces/user'
import UserList from './UserList'

export interface IProps {
  Users: IUser[]
  getAllUsers(): void
  selectedUserId: string
  selectUser(id: string): void
  addUser(id: string): void
}

const Users = (props: IProps) => {
  const [newUserName, setNewUserName] = useState('')
  const { Users, selectedUserId, selectUser, addUser, getAllUsers } = props
  const [error, setError] = useState('')

  const onAddUser = () => {
    if (newUserName !== '') {
      addUser(newUserName)
      setNewUserName('')
    } else {
      setError('Enter user name')
      return
    }
  }

  useEffect(() => {
    getAllUsers()
  }, [getAllUsers])
  return (
    <div>
      <div className={styles.errorDiv}>{error}</div>
      <div className={styles.addUserContainer}>
        <input
          type="text"
          id="newUser"
          placeholder="Enter user name"
          value={newUserName}
          onChange={({
            target: { value },
          }: React.ChangeEvent<HTMLInputElement>) => setNewUserName(value)}
          className={styles.searchContainerField}
        />
        <button
          className={styles.addUserbutton}
          onClick={onAddUser}
          id="addNewbutton"
        >
          Add User
        </button>
      </div>

      <UserList
        Users={Users}
        selectedUserId={selectedUserId}
        selectUser={selectUser}
      />
    </div>
  )
}

export default Users
