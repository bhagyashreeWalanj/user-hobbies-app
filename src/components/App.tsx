import * as React from 'react'
import Users from './../containers/UsersContainer'
import UserHobbies from './../containers/UserHobbyContainer'
import styles from './../styles/app.module.scss'
import { Provider } from 'react-redux'
import configureStore from '../store/Store'

const store = configureStore()

const App: React.FunctionComponent<{}> = () => {
  return (
    <Provider store={store}>
      <div className={styles.heading}>USER HOBBIES LIST</div>
      <div className={styles.container}>
        <Users />
        <UserHobbies />
      </div>
    </Provider>
  )
}

export default App
