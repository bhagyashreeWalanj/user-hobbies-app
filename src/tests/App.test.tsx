import App from '../components/App'
import Users from '../containers/UsersContainer'
import { shallow } from 'enzyme'
import UserHobby from '../containers/UserHobbyContainer'

describe('render all components', () => {
  const app = shallow(<App />)

  it('should match the snapshot', () => {
    expect(app).toMatchSnapshot()
  })
  it('App Component contains Users component', () => {
    expect(app.contains(<Users />)).toBe(true)
  })
  it('App Component contains Users Hobbies component', () => {
    expect(app.contains(<UserHobby />)).toBe(true)
  })
  it('should open User Hobbies Page without error', () => {
    expect(app.find('.heading').text()).toContain('USER HOBBIES LIST')
  })
})
