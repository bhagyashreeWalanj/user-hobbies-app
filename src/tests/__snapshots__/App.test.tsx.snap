// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render all components should match the snapshot 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <div
    className="heading"
  >
    USER HOBBIES LIST
  </div>
  <div
    className="container"
  >
    <Connect(Users) />
    <Connect(UserHobbies) />
  </div>
</Provider>
`;
